### ISSUE TITLE
Remove users of `get_unsound_ref_forever()`.

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#4861]
It produces `JSRef`s that are not actually rooted. There are two main users here:

* iterators in `node.rs`: I started in #4617, which is blocked on #4860;
* mutable roots in `treewalker.rs`: not sure yet what the solution is here; we'll probably have to refactor the code quite a bit.

**Comment by psdh** [comment]
I could try to fix this.

**Comment by jdm** [comment]
This will require fixing #4617 first (which requires (aka. is blocked on) #4860).

**Comment by doublec** [comment]
#4860 is now closed, but #4617 is marked as blocked on external. What is it blocked on? Is it the RootedVal mentioned in the critic review?

**Comment by jdm** [comment]
Yep.

**Comment by nox** [comment]
@Ms2ger Are you or someone else fixing the other iterators right now or can I do it?

**Comment by Ms2ger** [comment]
Go for it!

**Comment by nox** [comment]
I took care of some iterators: https://github.com/servo/servo/pull/5541.

**Comment by nox** [comment]
And TreeWalker: https://github.com/servo/servo/pull/5545

It sounds weird that the solution is that simple given you seem to imply it shouldn't.

**Comment by nox** [comment]
Node::collect_text_contents(): https://github.com/servo/servo/pull/5578

My fix for TreeWalker was wrong.

The remaining users of get_unsound_ref_forever() are:
* impl TNode for Node;
* TreeIterator;
* TreeWalker.

**Comment by jdm** [comment]
This was accomplished. Filing a new issue for the actual removal.
